NetCasbin/Model/FunctionMap.cs

```diff
@@ -17,15 +17,15 @@ public static FunctionMap LoadFunctionMap()
         {
             var map = new FunctionMap
             {
-                FunctionDict = new Dictionary<string, AbstractFunction>()
+                FunctionDict = new Dictionary<string, Delegate>()
             };
 
-            map.AddFunction("keyMatch", new KeyMatchFunc());
-            map.AddFunction("keyMatch2", new KeyMatch2Func());
-            map.AddFunction("keyMatch3", new KeyMatch3Func());
-            map.AddFunction("keyMatch4", new KeyMatch4Func());
-            map.AddFunction("regexMatch", new RegexMatchFunc());
-            map.AddFunction("ipMatch", new IPMatchFunc());
+            map.AddFunction("keyMatch",  (Func<string, string, bool>) BuiltInFunctions.KeyMatch);
```

Comment:
Can functions like `KeyMatchFunc()` be deleted now?

Reply:
These can be deleted in the library, but still can be used here now. I think we can retain these functions including AbstractFunction class for the user.
So this is not a breaking change, with no impact on the existing project.

Reply:
But you have changed the 2nd arg of AddFunction() from AbstractFunction to Delegate. If the user passes an old AbstractFunction type arg into the new version AddFunction(), I think this doesn't work?

Reply:
AbstractFunction has implemented the implicit operation of converting to Delegate. You can have a look at the latest commit.